src/common/buffer.cc

```diff
@@ -367,19 +377,15 @@ static ceph::spinlock debug_lock;
   ceph::unique_leakable_ptr<buffer::raw> buffer::create_small_page_aligned(unsigned len) {
     if (len < CEPH_PAGE_SIZE) {
       return create_aligned(len, CEPH_BUFFER_ALLOC_UNIT);
-    } else
+    } else {
       return create_aligned(len, CEPH_PAGE_SIZE);
+    }
   }
 
-  buffer::raw* buffer::create_unshareable(unsigned len) {
-    return new raw_unshareable(len);
+  ceph::unique_leakable_ptr<buffer::raw> buffer::create_unshareable(unsigned len) {
+    return ceph::unique_leakable_ptr<buffer::raw>(new raw_unshareable(len));
   }
 
-  buffer::ptr::ptr(raw* r) : _raw(r), _off(0), _len(r->len)   // no lock needed; this is an unref raw.
```

Comment:
This (and the `push_back()` as well) will require taking care of API compatibility which I'm going to cover with dedicated commit.